functions.php

```diff
@@ -1,5 +1,38 @@
 <?php
 
+/**
+*
+* Get a variable from a file and return its value; if not there, return the default.
+* NOTE: The variable's value is anything after the equal sign, so there shouldn't be a comment on the line.
+* NOTE: There may be something before $searchfor, e.g., "export X=1", where "X" is $searchfor.
+*/
+function get_variable($file, $searchfor, $default)
+{
+	// get the file contents
+	$contents = file_get_contents($file);
+	if ("$contents" == "") return($default);	// file not found or not readable
+
+	// escape special characters in the query
+	$pattern = preg_quote($searchfor, '/');
+	// finalise the regular expression, matching the whole line
+	$pattern = "/^.*$pattern.*\$/m";
```

Comment:
IMO... the `$pattern` should not allow for partial matches in the middle of the line.  Also, the `m` modifier ( multiline ) is not needed since we're only matching a single line for our options.

I think we should change the pattern to:
`"/^$pattern[ =].*$/"`

This will match only when the pattern begins the line, and is terminated by either a space or = sign, then anything from there to the end of line.

Let me know if I'm missing something...

Reply:
Michael, this function, and the identical one in allsky-portal, will go away once we have the unified config file since this function is only used to read config.sh and autocam.sh.  autocam.sh has a space before the variable name (export CAMERA=ZWO) so we can't search starting at the beginning of the line.
Also, since I have 2 cameras I have
~~~
CAMERA=ZWO
CAMERA=RPiHQ
~~~
in my config.sh file and I rely on get_variable() to use the second entry.  If I want to swithc the active camera I simply reverse the lines.

Can I ask that we leave these functions as is since they will be replaced shortly and the json file will resolve your concerns?